cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/mds/DatasetTypeMDS.java

```diff
@@ -32,91 +36,117 @@
 /**
  * Dataset types and modules metadata store
  */
-public class DatasetTypeMDS extends AbstractObjectsStore {
+public class DatasetTypeMDS extends MetadataStoreDataset {
   /**
    * Prefix for rows containing module info.
    * NOTE: we store in same table list of modules, with keys being <MODULES_PREFIX><module_name> and
    *       types to modules mapping with keys being <TYPE_TO_MODULE_PREFIX><type_name>
    */
-  private static final byte[] MODULES_PREFIX = Bytes.toBytes("m_");
+  private static final String MODULES_PREFIX = "m_";
 
   /**
    * Prefix for rows containing type -> module mapping
    * see {@link #MODULES_PREFIX} for more info.
    */
-  private static final byte[] TYPE_TO_MODULE_PREFIX = Bytes.toBytes("t_");
+  private static final String TYPE_TO_MODULE_PREFIX = "t_";
 
-  public DatasetTypeMDS(DatasetSpecification spec, @EmbeddedDataset("") OrderedTable table) {
-    super(spec, table);
+  public DatasetTypeMDS(DatasetSpecification spec, @EmbeddedDataset("") Table table) {
+    super(table);
   }
 
   @Nullable
-  public DatasetModuleMeta getModule(String name) {
-    return get(getModuleKey(name), DatasetModuleMeta.class);
+  public DatasetModuleMeta getModule(Id.DatasetModule datasetModuleId) {
+    return get(getModuleKey(datasetModuleId.getNamespaceId(), datasetModuleId.getId()), DatasetModuleMeta.class);
   }
 
   @Nullable
-  public DatasetModuleMeta getModuleByType(String typeName) {
-    String moduleName = get(getTypeKey(typeName), String.class);
-    if (moduleName == null) {
+  public DatasetModuleMeta getModuleByType(Id.DatasetType datasetTypeId) {
+    Id.DatasetModule datasetModuleId = get(getTypeKey(datasetTypeId.getNamespaceId(), datasetTypeId.getTypeName()),
+                                           Id.DatasetModule.class);
+    if (datasetModuleId == null) {
       return null;
     }
-    return getModule(moduleName);
+    // TODO: Slightly strange. Maybe change signature to accept Id.Namespace separately from typeName
+    return getModule(datasetModuleId);
   }
 
-  public DatasetTypeMeta getType(String typeName) {
-    DatasetModuleMeta moduleName = getModuleByType(typeName);
+  public DatasetTypeMeta getType(Id.DatasetType datasetTypeId) {
+    DatasetModuleMeta moduleName = getModuleByType(datasetTypeId);
     if (moduleName == null) {
       return null;
     }
-    return getTypeMeta(typeName, moduleName);
+    return getTypeMeta(datasetTypeId.getNamespace(), datasetTypeId.getTypeName(), moduleName);
```

Comment:
The ``getTypeMeta`` method does not always accept an ``Id.DatasetType``. It can accept ``(Id.Namespace, String typeName, DatasetModuleMeta)`` or ``(Id.Namespace, String typeName, Id.DatasetModule)``. Also, if you see the usage pattern else where, the namespace id and the typename are found separately (``getTypes`` method). So if I change the signature to accept an ``Id.DatasetType``, it would mean I would have to construct the ``Id.DatasetType`` when I have the namespace and the type name separately just for calling this method. 

Also, I feel that this signature calls out that the namespace belongs to both the module and the type.